src/NLog/MessageTemplates/ValueFormatter.cs

```diff
@@ -181,25 +204,37 @@ private bool SerializeSimpleObject(object value, string format, IFormatProvider
                 case TypeCode.UInt16:
                 case TypeCode.UInt32:
                 case TypeCode.UInt64:
-                {
-                    Enum enumValue;
-                    if ((enumValue = value as Enum) != null)
-                    {
-                        AppendEnumAsString(builder, enumValue);
-                    }
-                    else
                     {
-                        builder.AppendIntegerAsString(value, objTypeCode);
+                        Enum enumValue;
+                        if ((enumValue = value as Enum) != null)
```

Comment:
can we use `if(value is Enum enumValue)` here?

Reply:
Done